client/actions/fetchRecipe.js

```diff
@@ -0,0 +1,162 @@
+import { batchActions } from 'redux-batched-actions';
+
+import instance from '../config/axios';
+
+import { setFetching, unsetFetching } from './fetching';
+import { showNotification, hideNotification } from './notification';
+
+import {
+  RECIEVE_TOP_RATED_RECIPE,
+  RECIEVE_SINGLE_RECIPE,
+  RECIEVE_NEW_RECIPE,
+  RECIEVE_UPDATED_RECIPE
+} from './types';
+
+const recieveTopRecipes = (recipes, metaData) => ({
+  type: RECIEVE_TOP_RATED_RECIPE,
+  recipes,
+  metaData
+});
+
+const recieveSingleRecipe = recipe => ({
+  type: RECIEVE_SINGLE_RECIPE,
+  recipe
+});
+
+const recieveNewRecipe = recipe => ({
+  type: RECIEVE_NEW_RECIPE,
+  recipe
+});
+
+const recieveUpdatedRecipe = (recipe, index) => ({
+  type: RECIEVE_UPDATED_RECIPE,
+  recipe,
+  index
+});
+
+export const fetchSingleRecipes = id => (dispatch) => {
+  dispatch(setFetching());
+  return instance.get(`/recipes/${id}`)
+    .then((res) => {
+      dispatch(
+        batchActions([
+          recieveSingleRecipe(res.data.recipe),
+          unsetFetching()
+        ])
+      );
+    })
+    .catch((err) => {
+      console.log(err.message || err.response.data.message);
+      dispatch(unsetFetching());
+    });
+};
+
+export const fetchTopRecipes = (page, limit) => (dispatch) => {
+  dispatch(setFetching());
+  return instance.get(`/recipes?sort=upvotes&order=descending&page=${page}&limit=${limit}`)
+    .then((res) => {
+      dispatch(
+        batchActions([
+          recieveTopRecipes(res.data.recipes, res.data.metaData),
+          unsetFetching()
+        ])
+      );
+    })
+    .catch((err) => {
+      console.log(err.message || err.response.data.message);
+      dispatch(unsetFetching());
+    });
+};
+
+export const postReview = (content, id) => (dispatch) => {
+  dispatch(setFetching());
+  return instance.post(`/recipes/${id}/reviews`, { content })
+    .then((res) => {
+      dispatch(
+        batchActions([
+          recieveSingleRecipe(res.data.recipe),
+          unsetFetching()
+        ])
+      );
+    })
+    .catch((err) => {
+      console.log(err.response.data.message || err.message);
+      dispatch(unsetFetching());
+    });
+};
+
+export const vote = (dir, id) => dispatch => instance.post(`/recipes/${id}/vote-${dir}`)
+  .then((res) => {
+    dispatch(batchActions([
+      recieveSingleRecipe(res.data.recipe),
+      showNotification(res.data.message)
+    ]));
+    setTimeout(() => {
+      dispatch(hideNotification());
+    }, 3000);
+  })
+  .catch((err) => {
+    dispatch(showNotification(err.response.data.message));
```

Comment:
Are you sure you can depend on this error message to always be something sensible to the user?